GNUmakefile

```diff
@@ -0,0 +1,42 @@
+# This GNU make Makefile is intended as a helper.  meson is used for
+# the actual build and configuration, but this GNUmakefile provides
+# some useful targets for development and release purposes.  Type
+# "gmake help" for more details.  It is called GNUmakefile to indicate
+# it requires GNU make.
+
+# Allow calling environment overrides to commands used in this
+# GNUmakefile.  Defaults are below.
+
+REALPATH ?= realpath
+
+# Core directories
+MESON_BUILD_DIR = build
+topdir := $(shell $(REALPATH) $(dir $(lastword $(MAKEFILE_LIST))))
+
+# Project information (may be an easier way to get this from meson)
+PROJECT_NAME = $(shell grep ^project $(topdir)/meson.build | cut -d "'" -f 2)
+PROJECT_VERSION = $(shell grep version $(topdir)/meson.build | grep -E ',$$' | cut -d "'" -f 2)
+
+# Take additional command line argument as a positional parameter for
+# the GNUmakefile target
+TARGET_ARG = `arg="$(filter-out $@,$(MAKECMDGOALS))" && echo $${arg:-${1}}`
```

Comment:
These variables are not used anywhere. The code is not useful, especially when there is no meson.build in the source tree yet. Wouldn't you rather remove it from this commit, or wait with this commit after adding meson.build?